subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -0,0 +1,1547 @@
+				LOG_INF("TEST IS INTERRUPTING DOWNLOAD!");
+				k_sleep(K_SECONDS(2));
+				LOG_PANIC();
+				sys_reboot(0);
+			}
+#endif
+
+			if (pgps_need_assistance &&
+			    (finished || (index.loading_count > 1))) {
+				nrf_cloud_pgps_notify_prediction();
+			}
+
+			if (!finished) {
+				if (handler) {
+					handler(PGPS_EVT_LOADING, NULL);
+				}
+			} else {
+				LOG_INF("All PGPS data received. Done.");
+				state = PGPS_READY;
+				if (handler) {
+					handler(PGPS_EVT_READY, NULL);
+				}
+				npgps_print_blocks();
+				return 0;
+			}
+
+			index.store_block = npgps_alloc_block();
+			if (index.store_block == NO_BLOCK) {
+				LOG_ERR("No more free blocks!");
+				return -ENOMEM;
+			}
+			index.storage_extent--;
+			if (index.storage_extent == 0) {
+				index.storage_extent = npgps_get_block_extent(index.store_block);
+				LOG_INF("Moving to new flash region:%d, len:%d",
+					index.store_block, index.storage_extent);
+				err = flush_storage();
+				if (err) {
+					LOG_ERR("Error flushing storage:%d", err);
+					return err;
+				}
+				err = open_storage(npgps_block_to_offset(index.store_block),
+						   false);
+				if (err) {
+					LOG_ERR("Error opening storage again:%d", err);
+					return err;
+				}
+			}
+		}
+	} else {
+		LOG_ERR("Parsing incomplete; aborting.");
+		state = PGPS_NONE;
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int get_string_from_array(const cJSON *const array, const int index,
+				  char **string_out)
+{
+	__ASSERT_NO_MSG(string_out != NULL);
+
+	cJSON *item = cJSON_GetArrayItem(array, index);
+
+	if (!cJSON_IsString(item)) {
+		return -EINVAL;
+	}
+
+	*string_out = item->valuestring;
+
+	return 0;
+}
+
+static int parse_dl_info(char *host, size_t host_len,
+			 char *path, size_t path_len,
+			 const char *payload_in, size_t len)
+{
+	if (!host || !path || !payload_in) {
+		return -EINVAL;
+	}
+
+	char *temp;
+	char *host_ptr = NULL;
+	char *path_ptr = NULL;
+	int err = -ENOMSG;
+	cJSON *array = cJSON_Parse(payload_in);
+
+	if (!array || !cJSON_IsArray(array)) {
+		LOG_ERR("Invalid JSON array");
+		err = -EINVAL;
+		goto cleanup;
+	}
+
+	temp = cJSON_PrintUnformatted(array);
+	if (temp) {
+		LOG_DBG("JSON array: %s", log_strdup(temp));
+		cJSON_FreeString(temp);
+	}
+
+	if (get_string_from_array(array, RCV_ITEM_IDX_FILE_HOST, &host_ptr) ||
+	    get_string_from_array(array, RCV_ITEM_IDX_FILE_PATH, &path_ptr)) {
+		LOG_ERR("Error parsing info");
+		goto cleanup;
+	}
+
+	if (host_ptr && host_len) {
+		strncpy(host, host_ptr, host_len - 1);
+		host[host_len - 1] = '\0';
+		LOG_DBG("host: %s", log_strdup(host));
+	}
+	if (path_ptr && path_len) {
+		strncpy(path, path_ptr, path_len - 1);
+		path[path_len - 1] = '\0';
+		LOG_DBG("path: %s", log_strdup(path));
+	}
+	err = 0;
+
+cleanup:
+	if (array) {
+		cJSON_Delete(array);
+	}
+	return err;
+}
+
+/* handle incoming MQTT packets */
+int nrf_cloud_pgps_process(const char *buf, size_t buf_len)
+{
+	static char host[128];
+	static char path[64];
```

Comment:
use `CONFIG_DOWNLOAD_CLIENT_MAX_HOSTNAME_SIZE` and `CONFIG_DOWNLOAD_CLIENT_MAX_FILENAME_SIZE` ?
or at least swap swap the values, as path is usually > host